addons/website/static/src/client_actions/website_preview/website_preview.js

```diff
@@ -338,7 +338,33 @@ export class WebsitePreview extends Component {
         this.title.setParts({ action: currentTitle });
     }
 
-    _onPageLoaded() {
+    _onPageLoaded(ev) {
+        // FIX Chrome-only. If you have the backend in a language A but the
+        // website in English only, you can 1) modify a record's (event,
+        // product...) name in language A (say "New Name").
+        // 2) visit the page `/new-name-11` => the server will redirect you to
+        // the English page `/origin-11`, which is the only one existing.
+        // Chrome caches the redirection.
+        // 3) give the same name in English as in language A, try to visit
+        // => the server now wants to access `/new-name-11`
+        // => Chrome uses the cache to redirect `/new-name-11` to `/origin-11`,
+        // => the server tries to redirect to `/new-name-11` => start again 🔄.
+        // Chrome injects a "Too many redirects" layout in the iframe, which in
+        // turn raises a CORS error when the app tries to update the iframe.
+        // If we detect that behavior, we reload the iframe with a new query
+        // parameter, so that it's not cached for Chrome.
+        if (navigator.userAgent.toLowerCase().includes("chrome") && !this.iframeReloaded) {
+            try {
+                this.iframe.el.contentWindow.location.href;
+            } catch (err) {
+                if (err.name === "SecurityError") {
+                    ev.stopImmediatePropagation();
+                    this.iframe.el.src = `${this.iframe.el.src}${encodeURIComponent("?iframe_reload=1")}`;
```

Comment:
Url concat like that, in python or js is always a no go and red flag.
```js
this.$0.src
// https://54457432-16-0-all.runbot123.odoo.com/website/force/1?path=%2Fcontactus%3Frde%3Dtest%26role%3Dtest2
`${this.$0.src}${encodeURIComponent("?iframe_reload=1")}`
// https://54457432-16-0-all.runbot123.odoo.com/website/force/1?path=%2Fcontactus%3Frde%3Dtest%26role%3Dtest2%3Fiframe_reload%3D1
```

The URL when decoded turns into 
```
https://54457432-16-0-all.runbot123.odoo.com/website/force/1?path=/contactus?rde=test&role=test2?iframe_reload=1
```
instead of (mind the `&` instead of the `?` before `iframe_reload=1`)
```
https://54457432-16-0-all.runbot123.odoo.com/website/force/1?path=/contactus?rde=test&role=test2&iframe_reload=1
```

You may wanna do something like:
```js
const url = new URL(this.iframe.el.src);
url.searchParams.set('iframe_reload', 1)
```
Depending if you want to append the param to the url itself or the `?path=` url. It's probably better to have it on the "top level" URL (which this code does?) so it's not loaded in the real iframe URL (which we replicate in the browser URL), so it would be invisible to the user I believe.

Reply:
>  It's probably better to have it on the "top level" URL (which this code does?)

Actually, it's on the iframe URL (which is why I used `encodeURIComponent`), and it's needed because otherwise Chrome will see the same URL.

> ```js
> const url = new URL(this.iframe.el.src);
> url.searchParams.set('iframe_reload', 1)
> ```

I was under the impression that coming from the backend to a record in website, you wouldn't have any parameter. But yes, not robust enough.